### PR TITLE
added pd.Index with tests in common.pyi

### LINKED ISSUE #415: is_numeric_dtype not considering Index as a valid type
**Describe the bug**
As stated in the docs, the `is_numeric_dtype` function accepts Dtypes, Arrays, and Index as well.
https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_numeric_dtype.html?highlight=example

However, the typings are only contemplating `_ArrayOrDtype` 
https://github.com/pandas-dev/pandas-stubs/blob/001c0afb3ef0dfe21e96ced87c9e455c657a50c3/pandas-stubs/core/dtypes/common.pyi#L50

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
```python
from pandas.core.dtypes.common import is_numeric_dtype

is_numeric_dtype(pd.Index([1, 2.0]))


```
```
>>> mypy --version 
mypy 0.982 (compiled: yes)
```

Show the error message received from that type checker while checking your example.
(I put one type per line to make the output more readable)
```python
>>> mypy test.py     
/tmp/test.py:4: error: Argument 1 to "is_numeric_dtype" has incompatible type "Index"; 
expected 
"
Union[
    Union[ExtensionArray, ndarray[Any, Any]],
    Union[
        dtype[Any],
        None,
        Type[Any],
        _SupportsDType[dtype[Any]],
        str,
        Union[
            Tuple[Any, int],
            Tuple[Any, Union[SupportsIndex, Sequence[SupportsIndex]]],
            List[Any],
            _DTypeDict,
            Tuple[Any, Any],
        ],
    ],
    Series[Any],
    DataFrame,
]

"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

**Please complete the following information:**
OS: MacOS M1
OS Version Ventura 13.0
python version: 3.9.12
version of type checker: mypy 0.982 (compiled: yes)
version of installed pandas-stubs
pandas==1.5.1
pandas-stubs==1.5.1.221024

### PR DESCRIPTION
**Pull request by ramvikrams**
- [x] Closes #415 
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by ramvikrams** [comment]
Sir I had one doubt should the script test/nighly should be changed so that it's upgrades automatically for newer version of pip

**Comment by Dr-Irv** [comment]
> Sir I had one doubt should the script test/nighly should be changed so that it's upgrades automatically for newer version of pip

Just merged in #421 that fixes the nightly issue.  So you can merge with latest master to fix that.

File: tests/test_api_types.py [hunk 1]
```diff
@@ -18,46 +18,145 @@
...
 def test_is_array_like() -> None:
     check(assert_type(api.is_array_like(arr), bool), bool)
+    check(assert_type(api.is_array_like(np.array([1, 2, 3])), bool), bool)
+    check(assert_type(api.is_array_like(np.dtype(np.int32)), bool), bool)
+    check(assert_type(api.is_array_like(pd.Series([1, 2, 3])), bool), bool)
+    check(
+        assert_type(api.is_array_like(pd.DataFrame({"a": [1, 2], "b": [3, 4]})), bool),
+        bool,
+    )
+    check(assert_type(api.is_array_like(pd.Index([1, 2.0])), bool), bool)
```
Comment (Dr-Irv) [inline tests/test_api_types.py:28]: can you change all of these to use a module-level variable.  Right now, we have at the top of this file:
```python
nparr = np.array([1, 2, 3])
arr = pd.Series([1, 2, 3])
obj = "True"
mapping = {"a": "a"}
```
so you can add something for `np.dtype(np.int32)` , `(pd.DataFrame({"a": [1, 2], "b": [3, 4]})` and `pd.Index([1, 2.0])` up top, and reuse those variables throughout.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_api_types.py (hunk 1 above)
Comment (ramvikrams) [reply tests/test_api_types.py:28]: sure sir i'll do it

Comment (ramvikrams) [reply tests/test_api_types.py]: Done the changes sir

File: tests/test_api_types.py [hunk 2]
```diff
@@ -14,50 +14,131 @@
...
+        bool,
+    )
+    check(assert_type(api.is_bool_dtype(ind), bool), bool)
 
 
 def test_is_categorical_dtype() -> None:
     check(assert_type(api.is_categorical_dtype(arr), bool), bool)
+    check(assert_type(api.is_categorical_dtype(nparr), bool), bool)
+    check(assert_type(api.is_categorical_dtype(dtylike), bool), bool)
+    check(assert_type(api.is_categorical_dtype(arr), bool), bool)
```
Comment (Dr-Irv) [inline tests/test_api_types.py:57]: this is duplicate of a few lines above

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
One small change - you duplicated one test.  Otherwise looks good

**Comment by ramvikrams** [comment]
> One small change - you duplicated one test.  Otherwise looks good

Yes sir I'll fix it

**Comment by ramvikrams** [comment]
Done the change sir

**Review by Dr-Irv — APPROVED** [review]
Thanks @ramvikrams

**Comment by ramvikrams** [comment]
> Thanks @ramvikrams 

Thanks sir

**Pull request merged**